module3/src/MyBinarySearchTree.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by yurii.pyvovarenko on 11/28/2021.
+ * MyBinarySearchTree without balancing
+ */
+public class MyBinarySearchTree<T extends Comparable<T>> {
+    private static final String SPACE = " ";
+    private Node<T> root;
+
+    public MyBinarySearchTree(T value) {
+        root = new Node<T>(value);
+    }
+
+    void insert(T value)  {
+        root = insertRecursively(root, value);
+    }
+
+    Node insertRecursively(Node<T> root, T value) {
+        //tree is empty
+        if (root == null) {
+            root = new Node(value);
+            return root;
+        }
+        //traverse the tree
+        if (value.compareTo(root.getValue()) < 0) { //insert in the left subtree
```

Comment:
you are loosing the case, when `compareTo` return - 0. Binary tree should allow to store duplicates

Reply:
Corrected to `if (value.compareTo(root.getValue()) <= 0)`